manticore/platforms/linux.py

```diff
@@ -692,14 +692,23 @@ def __init__(
         # Keep track of the symbolic inputs we create
         self.inputs_recvd: List[ArrayProxy] = []
         self.recv_pos = 0
+        # This is a meta-variable, of sorts, and it is responsible for
+        # determining the symbolic length of the array during recv/read.
+        # Initially, it is None, to indicate we haven't forked yet. After
+        # fork, each state will be assigned their respective _actual_,
+        # concretized, receive length
+        self._symb_len: Optional[int] = None
+        # Set after adding this socket to the file descriptor list
+        self.fd: Optional[int] = None
 
     def __getstate__(self):
```

Comment:
```suggestion
    def __getstate__(self): -> State
```

Reply:
It's confusingly named, but `state` here (and in `__setstate__`) is actually a dict. `__getstate__` and `__setstate__` are pickle helpers, not something that's unique to Manticore, so the terminology collides a bit.

Reply:
Ahh ok. Good to know. These type hints are unnecessary then